fetcher/ubuntu.go

```diff
@@ -43,23 +45,23 @@ func FetchUbuntuVulnList() (entries []models.UbuntuCVEJSON, err error) {
 	}
 	log15.Debug(fmt.Sprintf("Ubuntu updated files: %d", len(targets)))
 
-	err = util.FileWalk(rootDir, targets, func(r io.Reader, _ string) error {
-		content, err := io.ReadAll(r)
-		if err != nil {
-			return err
-		}
+	return func(yield func(models.UbuntuCVEJSON, error) bool) {
+		var yeildErr = errors.New("yield error")
+		err = util.FileWalk(rootDir, targets, func(r io.Reader, _ string) error {
+			cve := models.UbuntuCVEJSON{}
+			if err = json.NewDecoder(r).Decode(&cve); err != nil {
+				return xerrors.Errorf("failed to decode Ubuntu JSON: %w", err)
+			}
 
-		cve := models.UbuntuCVEJSON{}
-		if err = json.Unmarshal(content, &cve); err != nil {
-			return xerrors.Errorf("failed to decode Ubuntu JSON: %w", err)
+			if !yield(cve, nil) {
+				return yeildErr
+			}
+			return nil
+		})
+		if errors.Is(err, yeildErr) { // No need to call yield with error
+			return
+		} else if err != nil && !yield(models.UbuntuCVEJSON{}, xerrors.Errorf("error in Ubuntu walk: %w", err)) {
+			return
```

Comment:
Why not first exclude the case where err is nil?
```suggestion
		if err != nil {
			if errors.Is(err, yeildErr) { // No need to call yield with error
				return
			}
			if !yield(models.UbuntuCVEJSON{}, xerrors.Errorf("error in Ubuntu walk: %w", err)) {
				return
			}
```